src/core/segment_reader.rs

```diff
@@ -393,3 +393,57 @@ impl<'a> Iterator for SegmentReaderAliveDocsIterator<'a> {
         result
     }
 }
+
+#[cfg(test)]
+mod test {
+    use core::{Index, SegmentReader};
+    use schema::*;
+
+    #[test]
+    fn test_iterator() {
+        let mut schema_builder = SchemaBuilder::new();
+        schema_builder.add_text_field("name", TEXT);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema.clone());
+        let name = schema.get_field("name").unwrap();
+
+        {
+            let mut index_writer = index.writer(50_000_000).unwrap();
```

Comment:
`index.writer_with_num_threads(1, 3_000_000)`...

`.writer(...)` is sniffing your number of cores to choose a number of threads and splits your heap accordingly. If the resulting heap is too small it will fail, so your test might not pass on a crazy high-end CPU.

On one thread, you can get away with using only 3MB.

Reply:
changed